index.js

```diff
@@ -394,8 +395,11 @@ Command.prototype.option = function(flags, description, fn, defaultValue) {
 
   // preassign default value only for --no-*, [optional], or <required>
   if (!option.bool || option.optional || option.required) {
-    // when --no-* we make sure default is true
-    if (!option.bool) defaultValue = true;
+    // when --no-foo we make sure default is true, unless a --foo option is already defined
+    if (!option.bool) {
+      var opts = self.opts();
+      defaultValue = name in opts ? opts[name] : true;
```

Comment:
Exploring alternatives to using plain `in`

Reply:
Sounds good! I'll use `Object.prototype.hasOwnProperty.call(opts, name)`.